submissions/Siusarna/data-structures/server/controllers.js

```diff
@@ -0,0 +1,118 @@
+const responseSender = require('./httpResponce.js');
```

Comment:
General logic left the same - your controller sends network responses. But now from wrapper.
Instead, controller should return result to place where it was called and basing on this result you should understand what happened - error or success.

It is called separation of concerns - controller don't care *who* and don't know *how* data is used. His only role is to *provide* data "on call".